duckduckgo/Android
# Add interstitial search

Task/Issue URL: https://app.asana.com/0/361428290920652/1154054879959297

**Description**:
Add a new interstitial search suggestion screen which displays autocomplete and device app results similar to that of the system default google app.

This interstitial screen is displayed when users launch our app
- via the google search bar, intent action com.duckduckgo.mobile.android.NEW_SEARCH
- DuckDuckGo widgets
- assistant (for users who set us as the system assistant)

**Steps to test this PR**:

**Test launch from Widget**
1. Run our app 
1. Install the widget
1. Tap the widget
1. Ensure the interstitial screen shows and widget launch pixel `m_w_l` fires

**Test launch from Assistant**
1. Run our app 
1. Set our app as the default Assistant
1. Press and hold the home button
1. Ensure the interstitial screen shows and the assistant launch pixel `m_a_l` fires

**Test launch from Google action**
1. Close our app
1. From the terminal run `adb shell am start -a com.duckduckgo.mobile.android.NEW_SEARCH` to send the correct action to our app
1. Ensure the interstitial screen shows and google bar launch pixel `m_g_l` fires

**Test Dax icon opens main app**
1. Open the interstitial screen again
1. Tap on the dax icon
1. Ensure this launches our app and the interstitial app launch pixel `m_i_ld` fires

**Test autocomplete selection opens entry in browser**
1. Open the interstitial screen again
1. Start typing
1. Select an autcomplete result
1. Ensure this launches in our browser and the interstitial browser launch pixel `m_i_lbq` fires

**Test app selection entry opens app**
1. Open the interstitial screen again
1. Start typing `Goo`
1. Select an app result
1. Ensure this launches the app and the interstitial device app pixel `m_i_sda` fires

**Test UI**
1. Check the interstitial screen in landscape and split screen

**Test Migration**
1. Uninstall previous versions of the app
1. Install develop version of the app, run it and add widget
1. Install this version over the top
1. Tap widget to ensure it still works

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
